dagjose/dagjose_assembler.go

```diff
@@ -155,26 +156,11 @@ func (d *dagJOSENodeBuilder) AssignLink(l ipld.Link) error {
 	return dagJoseMixin.AssignLink(l)
 }
 func (d *dagJOSENodeBuilder) AssignNode(n ipld.Node) error {
-	if d.state != maState_initial {
-		panic("misuse")
-	}
-	if n.Kind() != ipld.Kind_Map {
-		return ipld.ErrWrongKind{TypeName: "map", MethodName: "AssignNode", AppropriateKind: ipld.KindSet_JustMap, ActualKind: n.Kind()}
-	}
-	itr := n.MapIterator()
-	for !itr.Done() {
-		k, v, err := itr.Next()
-		if err != nil {
-			return err
-		}
-		if err := d.AssembleKey().AssignNode(k); err != nil {
-			return err
-		}
-		if err := d.AssembleValue().AssignNode(v); err != nil {
-			return err
-		}
+	err := datamodel.Copy(n, d)
```

Comment:
I can't see this function on `master` in `go-ipld-prime`, is this PR dependent on some pending changes to `go-ipld-prime`?

Also, this could just be `return datamodel.Copy(n, d)` right?

Reply:
Yes, you're right about the code change - will update.

I'll chat with @oed and see how to proceed with releasing this. Worst case, I could clone @warpfork's code with credit and a TODO to use the official go-ipld-prime code once it is released.

Reply:
Here's Eric's PR:
https://github.com/ipld/go-ipld-prime/pull/264